### DISCUSSION TITLE
config.kit.target is no longer required, and should be removed

### DISCUSSION THREAD

**Discussion by davehorner** [sveltejs/kit#3695] [Q&A] [answered]
### Describe the bug

```
pnpm run sdev

> c@1.0.0 sdev C:\Users\dhorner\work\tauri\c
> svelte-kit dev

> config.kit.target is no longer required, and should be removed
    at file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:874:10
    at file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:766:43
    at file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:752:18
    at file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:752:18
    at validate_config (file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:936:9)
    at load_config (file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:913:20)
    at async file:///C:/Users/dhorner/work/tauri/c/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.259_svelte@3.46.3/node_modules/@sveltejs/kit/dist/cli.js:1013:19
 ELIFECYCLE  Command failed with exit code 1.
```

### Reproduction

https://github.com/tauri-apps/tauri/tree/dev/tooling/create-tauri-app creates a package.json that has sdev and sbuild.  I was trying to run them on my machine.  pnpm run sdev

config.kit.target is no longer required, and should be removed
I don't know how to remove this setting.

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 15.14.0 - C:\ProgramData\Miniconda3\node.EXE
    Yarn: 1.22.10 - C:\ProgramData\Miniconda3\yarn.CMD
    npm: 7.19.1 - C:\ProgramData\Miniconda3\npm.CMD
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.17
    @sveltejs/kit: next => 1.0.0-next.259
    svelte: ^3.44.0 => 3.46.3
```

### Severity

blocking all usage of SvelteKit

### Additional Information

config.kit.target is no longer required, and should be removed

**Comment by ignatiusmb** [comment, marked as answer]
As the error says, remove `target` option from the config

**Reply by IamSheikh** [reply]
Thank you sooo much

**Comment by nberlette** [comment]
@davehorner try removing the `target` option under `kit` in your project's root `svelte.config.js`.

Maybe the error message should clarify that it's talking about `svelte.config.js`? The `config.kit.target` path might be obvious to seasoned users, but with no reference to a filename or error stack, many users might be confused on exactly **which** config file needs amending.

P.S. wouldn't it be... nicer to spit out a deprecation warning - but continuing to build - rather than a hard failure? SvelteKit has done this before with many other methods / properties. Seems odd to change the pattern now.

**Reply by davehorner** [reply]
I think if "svelte.config.js" was added to that message it would be clear.  I'm new :)
---- 
still running into other errors.  looks like the code generated is old from tauri.

request in handle has been replaced with event. See https://github.com/sveltejs/kit/pull/3384 for details                                                                                     Error: request in handle has been replaced with event. See https://github.com/sveltejs/kit/pull/3384 for details 

$page.path has been replaced by $page.url.pathname                                                                                                                                            Error: $page.path has been replaced by $page.url.pathname

**Comment by Yakumwamba** [comment]
Hi , so I have a question regarding the removal of the `config.kit.target`, how does SvelteKit  hydrate the <div id="svelte"> element in src/app.html? 

Thanks, Im new here.

**Reply by xpat** [reply]
@Yakumwamba This may be the answer to your question:  https://github.com/sveltejs/kit/pull/3674

> use data-hydrate instead of explicit target  [https://github.com/sveltejs/kit/pull/3674/commits/f9d529fc558e34e3842af0f28da8e1b106c23644](url)

**Reply by Yakumwamba** [reply]
Hey xpat, Thanks for the anwer. I removed `config.target` and I have no errors
